platform/openide.util.ui/src/org/openide/util/Utilities.java

```diff
@@ -1325,6 +1325,32 @@ private static Rectangle findCenterBounds(GraphicsConfiguration gconf, Dimension
         );
     }
 
+    /**
+     * This is for use in situations where a standard swing API,
+     * such as {@linkplain JOptionPane.show*} or {@linkplain JFileChooser.show*},
+     * is used to display a dialog. {@code null} should never be used
+     * as a dialog's parent because it
+     * frequently does the wrong thing in a multi-screen setup.
+     * <p>
+     * The use of the NetBeans API
+     * {@linkplain DialogDisplayer.getDefault.*}
+     * is encouraged to display a dialog, but stuff happens.
+     * @return A suitable parent component for swing dialog displayers.
+     * @since 9.26
+     */
+    // PR4739
+    public static Component findDialogParent() {
+        Component parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
+        if (parent == null) {
+            parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getActiveWindow();
+        }
+        if (parent == null) {
+            Frame[] f = Frame.getFrames();
+            parent = f.length == 0 ? null : f[f.length - 1];
```

Comment:
The documentation of `Frame.getFrames` reads a bit strange. For NetBean I suggest to fall back to `WindowManager.getMainWindow` and if that indeed is not present, the return null.

Reply:
> The documentation of `Frame.getFrames` reads a bit strange. For NetBean I suggest to fall back to `WindowManager.getMainWindow` and if that indeed is not present, the return null.

Strange indeed. `getMainWindow()` does seem a better fallback. (and acc'd to doc creates a frame if none exists.)

I'll make that change, **please review carefully since I won't be able to test it**.

Reply:
This is code that already exists in the platform, in FileChooserBuilder. I'm not sure using getMainWindow can happen without causing dependency issues?

Reply:
I'm manually looking for main window specifically to avoid adding the dependency. Similar to whats seen around 
https://github.com/apache/netbeans/blob/abb1a97c84e22da99cd38640d52f57e029fa2dbb/platform/openide.util.ui/src/org/openide/util/Utilities.java#L1160-L1164

Reply:
So I'll wait for further comment. I've got a change ready that find the main window (without a dependency). 

I **will not push it**, unless/until there's a consensus.

Reply:
Oh  yeah, it looks like
```
        // KFM is out of it, check out known frames
        if (parent == null) {
            Frame[] frames = Frame.getFrames();
            for( Frame f : frames ) {
                if( "NbMainWindow".equals(f.getName())) { //NOI18N
                    parent = f;
                    break;
                }
            }
            if(parent == null && frames.length != 0)
                parent = frames[frames.length - 1];
        }
```

Reply:
I am strongly inclined to leave the code as a verbatim move of the code from `FileChooserBuilder` for now - that is used a lot across the IDE already.  Let's use as is for the additional cases unless there's a proven issue to address.

While the documentation `Frame::getFrames` is a bit odd, I'm curious how often that fallback is ever hit here?  Have you tried a breakpoint?  If it's rare/never then changing it is pointless.  If it's commonly hit, then special casing `NbMainWindow` may be a change in logic that actually ends up being less correct.

Reply:
There's no consensus to change. **As is.**